ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -718,33 +718,40 @@ func (r *RayServiceReconciler) updateServeDeployment(ctx context.Context, raySer
 	}
 }
 
-// getAndCheckServeStatus gets Serve applications' and deployments' statuses,
-// updates health timestamps, and checks if the RayCluster is overall healthy.
-// It takes as one of its inputs `serveConfigType`, which is used to decide whether
-// to query the single-application Serve REST API or the multi-application Serve REST API
-// It's return values should be interpreted as
-// (Serve app healthy?, Serve app ready?, error if any)
+// `getAndCheckServeStatus` gets Serve applications' and deployments' statuses, updates health timestamps,
+// and checks if the RayCluster is overall healthy. It takes as one of its inputs `serveConfigType`, which
+// is used to decide whether to query the single-application Serve REST API or the multi-application Serve
+// REST API. It's return values should be interpreted as:
+//
+// (Serve app healthy?, Serve app ready?, error if failed to get Serve statuses)
 func (r *RayServiceReconciler) getAndCheckServeStatus(ctx context.Context, dashboardClient utils.RayDashboardClientInterface, rayServiceServeStatus *rayv1alpha1.RayServiceStatus, serveConfigType utils.RayServeConfigType, unhealthySecondThreshold *int32) (bool, bool, error) {
-	// If the unhealthySecondThreshold value is non-nil, then we will use that value.
-	// Otherwise, we will use the value ServiceUnhealthySecondThreshold which can be set in a test
-	// This is used for testing purposes.
+	// If the `unhealthySecondThreshold` value is non-nil, then we will use that value. Otherwise, we will use the value ServiceUnhealthySecondThreshold
+	// which can be set in a test. This is used for testing purposes.
 	serviceUnhealthySecondThreshold := ServiceUnhealthySecondThreshold
 	if unhealthySecondThreshold != nil {
 		serviceUnhealthySecondThreshold = float64(*unhealthySecondThreshold)
 	}
 
+	// TODO (kevin85421): Separate the logic for retrieving Serve application statuses from checking Serve application statuses into two separate functions.
+	// Currently, the handling logic for `isHealthy` and `isReady` between these two behaviors is inconsistent. This can cause potential issues in the future.
 	var serveAppStatuses map[string]*utils.ServeApplicationStatus
 	var err error
 	if serveConfigType == utils.SINGLE_APP {
 		var singleApplicationStatus *utils.ServeApplicationStatus
 		if singleApplicationStatus, err = dashboardClient.GetSingleApplicationStatus(ctx); err != nil {
-			r.Log.Error(err, "Failed to get Serve deployment statuses from dashboard!")
+			err = fmt.Errorf(
+				"Failed to get Serve deployment statuses from the dashboard agent (port: 52365). "+
```

Comment:
```suggestion
				"Failed to get Serve deployment statuses from the dashboard agent (port: 52365). "+
```
Is the port configurable? If it's easy to include the actual port here, we should, instead of hardcoding it.

Reply:
Good idea. Would it be better if I update the error message to "Failed to retrieve Serve deployment statuses from the dashboard agent port of the head service (`dashboard-agent`)"? The actual port will be shown in `err`. In Issue 7, the error message is `Get \"http://rayservice-sample-raycluster-rqlsl-head-svc.default.svc.cluster.local:52365/api/serve/applications/\": dial tcp 10.96.7.154:52365: connect: connection refused`.

Reply:
Updated https://github.com/ray-project/kuberay/pull/1230/commits/107edd327906cf5868bdd3cfa6ba560c98c7ba1c

Reply:
Sounds good!